tests/test_time_series/test_time_lag.py

```diff
@@ -0,0 +1,150 @@
+import numpy as np
+import pandas as pd
+import pytest
+
+from feature_engine.timeseries.forecasting import TimeSeriesLagTransformer
+_false_input_params = [
+    ([3, 2], "pizza", ["tango", "empanada"], 9),
+    ("mate", ["bombilla", "limpiado"], True, "cocinado"),
+    (True, False, "asado", [1984, 1999]),
+]
+
+
+@pytest.mark.parametrize(
+    "_variables, _periods, _freq, _drop_original",
+    _false_input_params
+)
+def test_raises_error_when_wrong_input_params(
+        _variables, _periods, _freq, _drop_original, df_time
```

Comment:
Some of these params fail straightaway in __init__, for example `variables` and `drop_original`, and some fail during transform(), like the params that are used by pandas (period, freq).

I am not sure if we need a test, to test that pandas fails when entering the wrong params, pandas should have a check already. @KishManani what is your view on this?

Regarding our params that should fail in the init (variables and drop_original), can we test them in their own test?

Also, note that [3,2] is a valid input for variables in init. 

I suggest we either split the test in 2, one with the param values that fail in fit and one with those that fail in the init.

Reply:
I agree that a test here is not necessary for the reasons you've provided.

Reply:
Ok, so the plan is to only test `variables` and `drop_original`? In that case, should we delete the `fit()` and `transform()` methods?

Reply:
@Morgan-Sell 

to test the variables param, we have now a generic test that we can apply to all transformers. The test is [here](https://github.com/feature-engine/feature_engine/blob/main/feature_engine/estimator_checks.py#L102).

We also have [this other test](https://github.com/feature-engine/feature_engine/blob/main/feature_engine/estimator_checks.py#L31) that runs the above and other multiple tests that we need to run on all our transformers.

And here you have [an example](https://github.com/feature-engine/feature_engine/blob/main/tests/test_selection/test_check_estimator_selectors.py#L46) on how to run those tests for all selector transformers. For this module, at the moment we have only 1 class, but we will have more. We should do the same.

In short, we need a script like the test_check_estimator_selectors for the forecasting module, that runs the check_estimator test from sklearn (first test in the script) and the check_estimator from feature engine (second test in this script).

If you are feeling dareful, we could also create a check for the param drop_original in the estimator_checks and run it from the master check scropt. Because other transformers also have this parameter, and would benefit from this test.

Reply:
I am feeling intrepid!  I'll embark on this adventure later ;)

Reply:
Ok, well, it seems like someone already developed the sklearn and feature-engine check_estimator tests.

A couple of questions about the `drop_original` param check:

- Are you envisioning a check function that assesses whether `drop_original` is a boolean variable? Is it doing anything else?
- What/where is the master check script?

Reply:
yes, that was me, lol.

the check needs to check that if it is set to false, the original variables are retained, and if set to true, the original variables are dropped after the transformation.

I wonder if we should also check within the same test, that it takes the allowed param. have a think and let me know your views :)

Reply:
Ok, I understand the desired test. Are you thinking that we use include that test in `test_different_types_for_init_params()`?

If so, it seems better to create its own test b/c `test_different_types_for_init_params()` does not require `fit()` and `transform()`. I assume it would be cleaner for someone to understand what's occurring if it were a separate unit test.